core/src/main/java/org/mskcc/cbio/portal/model/CanonicalGene.java

```diff
@@ -56,19 +57,39 @@
     private String type;
 
     public CanonicalGene(String hugoGeneSymbol) {
-        this(-1, hugoGeneSymbol);
-    }
-
-    public CanonicalGene(long entrezGeneId, String hugoGeneSymbol) {
-        this(entrezGeneId, hugoGeneSymbol, null);
+        this(hugoGeneSymbol, null);
     }
 
     public CanonicalGene(String hugoGeneSymbol, Set<String> aliases) {
-        this(-1, hugoGeneSymbol, aliases);
```

Comment:
these overloaded constructors are not easy to comprehend on first reading .. I would recommend some javadoc comments to explain the logic of these constructors (use cases). Alternatively, we could try to make the constructors simpler / fewer.
postscript: I see there are javadoc comments below here, and those are helpful